ScnSoft-Education/LeetCodeTraining/ZigzagConversion/ZigzagConversionSolution.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace LeetCodeTraining.ZigzagConversion
+{
+    //https://leetcode.com/problems/zigzag-conversion/
+    public static class ZigzagConversionSolution
+    {
+        public static string Convert(string s, int numRows)
+        {
+            if (s.Length <= numRows || numRows == 1)
+            {
+                return s;
+            }
+
+            char[] result = new char[s.Length];
+            int lookupIndex;
+            int counter = 0;
+            bool fromTop; // responsible for the way of calculating the next char
+
+            for (int row = 0; row < numRows; row++)
+            {
+                lookupIndex = row;
+                fromTop = true;
+
+                while (true)
+                {
+                    result[counter] = s[lookupIndex];
+
+                    CalculateIndeces(ref counter, ref lookupIndex, ref fromTop, row);
```

Comment:
It seems passing counter, lookupIndex, fromTop is not required for nested function and can be deleted.